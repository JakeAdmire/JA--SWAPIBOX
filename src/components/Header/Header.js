import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { setCategory } from '../../actions';
import { Button } from '../Button/Button';

export function Header(props) {
  let faves = `Faves(${props.faves.length})`;

  const passEvent = (e) => {
    const { innerText } = e.target;
    props.setCategory(innerText);
  }

  const generateButtons = () => {
    const buttons = ['People', 'Planets', 'Vehicles'];
    return buttons.map(button => 
      <Button key={button} button={button} passEvent={passEvent} />
    )
  }

  return (
    <div>
      <h1>Swapi-Box</h1>
      <div className="button-holder">
        { generateButtons() }
      </div>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  faves: state.faves
})

export const mapDispatchToProps = (dispatch) => ({
  setCategory: (event) => dispatch(setCategory(event))
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);