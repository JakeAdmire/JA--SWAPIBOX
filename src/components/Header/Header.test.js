import { enzyme, shallow } from 'enzyme';
import React from 'react'

import { Header, mapStateToProps } from './Header';

describe('Header', () => {

  let wrapper;
  let setCategory = jest.fn();

  wrapper = shallow(
    <Header faves={0} setCategory={setCategory} />
  )

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }) 

  it.skip('should invoke passEvent() on button clicked', () => {
    let e = {target: {innerText: 'People'}};
    wrapper.find('.people').simulate('click', e);
    expect(wrapper.passEvent).toBeCalledWith(e);
  })

})

describe('mapStateToProps', () => {

  it('should return a faves array as props', () => {
    const mockState = {
      faves: []
    }

    const results = mapStateToProps(mockState)
    expect(results).toEqual(mockState)
  })

})