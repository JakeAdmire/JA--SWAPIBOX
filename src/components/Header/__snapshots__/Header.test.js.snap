// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    faves={0}
    setCategory={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Swapi-Box
        </h1>,
        <div
          className="button-holder"
        >
          <Button
            button="People"
            passEvent={[Function]}
          />
          <Button
            button="Planets"
            passEvent={[Function]}
          />
          <Button
            button="Vehicles"
            passEvent={[Function]}
          />
          <button>
            Faves(undefined)
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Swapi-Box",
        },
        "ref": null,
        "rendered": "Swapi-Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <Button
                button="People"
                passEvent={[Function]}
              />,
              <Button
                button="Planets"
                passEvent={[Function]}
              />,
              <Button
                button="Vehicles"
                passEvent={[Function]}
              />,
            ],
            <button>
              Faves(undefined)
            </button>,
          ],
          "className": "button-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "People",
            "nodeType": "function",
            "props": Object {
              "button": "People",
              "passEvent": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Planets",
            "nodeType": "function",
            "props": Object {
              "button": "Planets",
              "passEvent": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Vehicles",
            "nodeType": "function",
            "props": Object {
              "button": "Vehicles",
              "passEvent": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Faves(undefined)",
            },
            "ref": null,
            "rendered": "Faves(undefined)",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Swapi-Box
          </h1>,
          <div
            className="button-holder"
          >
            <Button
              button="People"
              passEvent={[Function]}
            />
            <Button
              button="Planets"
              passEvent={[Function]}
            />
            <Button
              button="Vehicles"
              passEvent={[Function]}
            />
            <button>
              Faves(undefined)
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swapi-Box",
          },
          "ref": null,
          "rendered": "Swapi-Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <Button
                  button="People"
                  passEvent={[Function]}
                />,
                <Button
                  button="Planets"
                  passEvent={[Function]}
                />,
                <Button
                  button="Vehicles"
                  passEvent={[Function]}
                />,
              ],
              <button>
                Faves(undefined)
              </button>,
            ],
            "className": "button-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "People",
              "nodeType": "function",
              "props": Object {
                "button": "People",
                "passEvent": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Planets",
              "nodeType": "function",
              "props": Object {
                "button": "Planets",
                "passEvent": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Vehicles",
              "nodeType": "function",
              "props": Object {
                "button": "Vehicles",
                "passEvent": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Faves(undefined)",
              },
              "ref": null,
              "rendered": "Faves(undefined)",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
